_plugins/sidebar.rb

```diff
@@ -0,0 +1,216 @@
+# frozen_string_literal: false
+
+require 'jekyll'
+require 'nokogiri'
+require 'json'
+
+###
+# This plugin works only if used as a remote-theme
+# If included as a theme or gem this plugin must be copied over
+###
+
+module Jekyll
+  class Sidebar
+    attr_accessor :hash_pre_render
+    attr_accessor :filename_with_headers
+
+    def initialize
+      @hash_pre_render = {}
+      @filename_with_headers = {}
+    end
+
+    def pre_render(page)
+      menu_order = page['menu-order'].nil? ? 0 : page['menu-order']
+      hide_from_sidebar = page['hide_from_sidebar'].nil? ? false : page['hide_from_sidebar']
+      url = page['url'].gsub('index.html', '').gsub('.html', '')
+      @hash_pre_render[url] = {
+        title: page['title'],
+        url: page['url'].gsub('.html', ''),
+        name: page['name'],
+        menu_order: menu_order,
+        hide_from_sidebar: hide_from_sidebar
+      }
+    end
+
+    def post_write(site)
+      files = []
+      Dir.glob("#{site.config['destination']}/**/*.html") do |filename|
+        doc = File.open(filename) { |f| Nokogiri::HTML(f) }
+        files.push(doc)
+
+        headers = []
+        doc.xpath('//h2 ').each do |header|
+          next unless header['id']
+
+          child = header.last_element_child
+          header = {
+            id: header['id'],
+            title: header.content.strip,
+            hash: (child['href']).to_s
+          }
+          headers.push(header)
+        end
+        sanitized_filename = sanitize_filename(filename)
+        @filename_with_headers[sanitized_filename] = { headers: headers }
+      end
+
+      Dir.glob("#{site.config['destination']}/**/*.html") do |filename|
+        sanitized_filename = sanitize_filename(filename)
+        sidebar = render(sanitized_filename)
+        file = File.open(filename) { |f| Nokogiri::HTML(f) }
+        file.xpath('//*[@id="dx-sidebar-main-nav-ul"]').each do |location|
+          location.inner_html = sidebar
+        end
+        File.open(filename, 'w') { |f| f.write(file.to_html(encoding: 'UTF-8')) }
+      end
+
+      # File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar) }
+    end
+
+    private
+
+    def sanitize_filename(filename)
+      sanitized_filename = filename.match(/(?m)(?<=\b_site).*$/)[0]
+      sanitized_filename = sanitized_filename.gsub('index.html', '')
+      sanitized_filename.gsub('.html', '')
+    end
+
+    def generateSubgroup(filename, key, value, all_subgroups, level)
```

Comment:
Codacy found an issue: [Perceived complexity for generateSubgroup is too high. [16/7]](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6002908)